src/UserInterface/CarMap.py

```diff
@@ -111,15 +151,20 @@ def check_vehicle_proximity(self, vehicle_id: str, position: dict,) -> None:
         position : dict
             Dictionary containing the 'x' and 'y' coordinates of the vehicle's position in the simulation.
         """
-        for v in self._environment_manager._active_virtual_cars:
-            if v == vehicle_id:
-                self.check_virtual_vehicle_proximity(vehicle_id, position)
+        if self.__can_virtual_cars_hack is True:
+            for v in self._environment_manager._active_virtual_cars:
+                if v == vehicle_id:
+                    self.check_hack_vehicle_proximity(vehicle_id, position)
+        if self.__can_physical_cars_hack is True:
+            for v in self._environment_manager._active_physical_cars:
+                if v == vehicle_id:
+                    self.check_hack_vehicle_proximity(vehicle_id, position)
```

Comment:
To check wether or not a vehicle id is in the virtual or physical list you could just write:

if vehicle_id in self._environment_manager._active_virtual_cars

But you should not access private attributes of other classes anyway. There should be a public method in EnvironmentManager to check wether or not a vehicle is registered as virtual or physical.